tests/sub_group/sub_group_api.cpp

```diff
@@ -0,0 +1,368 @@
+              accessor_value[idx] = sub_group.leader();
+              ASSERT_RETURN_TYPE(bool, sub_group.leader(),
+                                 "sycl::sub_group.leader()");
+            }
+            {  // group.get_group_linear_id()
+              const size_t idx = get_idx(lgid, method::group_group_linear_id);
+              accessor_value[idx] = item.get_group_linear_id();
+              // method belongs to sycl::group, no return type check needed
+            }
+          });
+    });
+  }
+
+  // check that all elements were assigned a value
+  REQUIRE(std::all_of(results.begin(), results.end(),
+                      [=](size_t val) { return val != special_val; }));
+
+  // max_local_range of first work-item, used to check if they are all the same
+  size_t first_max_local_range = results[get_idx(0, method::max_local_range)];
+  // check isolated results for each work-item
+  for (size_t i = 0; i < linear_execution_range; i++) {
+    INFO("index: " << i);
+    const auto get = [=](method method) -> size_t {
+      return results[get_idx(i, method)];
+    };
+
+    // group_id, group_linear_id
+    size_t idx_group_id = get(method::group_id);
+    CHECK((idx_group_id == get(method::group_linear_id)));
+    CHECK((idx_group_id < get(method::group_linear_range)));
+
+    // local_id, local_linear_id
+    size_t idx_local_id = get(method::local_id);
+    CHECK((idx_local_id == get(method::local_linear_id)));
+    CHECK((idx_local_id < get(method::local_range)));
+
+    // local_range
+    size_t idx_local_range = get(method::local_range);
+    CHECK((idx_local_range == get(method::local_linear_range)));
+    CHECK((idx_local_range <= get(method::max_local_range)));
+
+    // group_range
+    CHECK((get(method::group_range) == get(method::group_linear_range)));
+
+    // leader: check that true iff local id is zero
+    CHECK((get(method::leader) == (get(method::local_id) == 0)));
+
+    // max_local_range: check that every work-item returns the same value
+    CHECK((first_max_local_range == get(method::max_local_range)));
+  }
+
+  // max_local_range: check that it is a valid sub-group size
+  const std::vector<size_t> valid_sub_group_sizes =
+      device.get_info<sycl::info::device::sub_group_sizes>();
+  CHECK(
+      std::any_of(valid_sub_group_sizes.begin(), valid_sub_group_sizes.end(),
+                  [=](size_t size) { return size == first_max_local_range; }));
+
+  // work-items are assigned to sub-groups in an implementation-defined way,
+  // and sub_group.get_group_id() cannot be used to identity work-items in the
+  // execution range. the below checks verify properties without depending on
+  // implementation details. note therefore that "global index" as used below
+  // is calculated rather than expected based on the work-item id
+
+  // check that all work-items in the same work-group have the same number
+  // of sub-groups. use REQUIRE as sub_group_counts is used by next checks
```

Comment:
```suggestion
  // Work-items are assigned to sub-groups in an implementation-defined way,
  // and sub_group.get_group_id() cannot be used to identity work-items in the
  // execution range. The checks below verify properties without depending on
  // implementation details. Note therefore that "global index" as used below
  // is calculated rather than expected based on the work-item id.

  // Check that all work-items in the same work-group have the same number
  // of sub-groups. Use REQUIRE as sub_group_counts is used by next checks
```